R/utilities-output-selections.R

```diff
@@ -17,24 +20,32 @@
 #' parameter <- getParameter("Organism|Liver|Volume", sim)
 #' addOutputs(parameter, sim)
 #' @export
-addOutputs <- function(quantitiesOrPaths, simulation) {
+addOutputs <- function(quantitiesOrPaths, simulation, returnQuantities = TRUE) {
   quantitiesOrPaths <- c(quantitiesOrPaths)
 
   validateIsOfType(quantitiesOrPaths, c(Quantity, "character"))
   validateIsOfType(simulation, Simulation)
-  quantities <- quantitiesOrPaths
 
-  if (isOfType(quantitiesOrPaths, "character")) {
-    quantities <- getAllQuantitiesMatching(quantitiesOrPaths, simulation)
+  # If quantities are provided, get their paths
+  paths <- vector("character", length(quantitiesOrPaths))
+  if (isOfType(quantitiesOrPaths, Quantity)) {
+    for (idx in seq_along(quantitiesOrPaths)) {
+      paths[[idx]] <- quantitiesOrPaths[[idx]]$path
+    }
+  } else {
+    paths <- quantitiesOrPaths
   }
+  paths <- unique(paths)
 
-  quantities <- uniqueEntities(quantities, compareBy = "path")
-  outputSelections <- simulation$outputSelections
-
-  for (quantity in quantities) {
-    outputSelections$addQuantity(quantity)
+  task <- ospsuite:::getContainerTask()

```

Comment:
why do you use the namespace method within the package? I'd remove this

Reply:
Oops, must be copy-paste.. Will remove